pkg/operator/ceph/object/notification/controller.go

```diff
@@ -0,0 +1,257 @@
+	client      client.Client
+	scheme      *runtime.Scheme
+	context     *clusterd.Context
+	clusterInfo *cephclient.ClusterInfo
+	clusterSpec *cephv1.ClusterSpec
+}
+
+// Add creates a new ObjectBucketClaim Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager, context *clusterd.Context) error {
+	return add(mgr, newReconciler(mgr, context))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager, context *clusterd.Context) reconcile.Reconciler {
+	// Add the cephv1 scheme to the manager scheme so that the controller knows about it
+	mgrScheme := mgr.GetScheme()
+	if err := cephv1.AddToScheme(mgr.GetScheme()); err != nil {
+		panic(err)
+	}
+	return &ReconcileOBCLabels{
+		client:  mgr.GetClient(),
+		scheme:  mgrScheme,
+		context: context,
+	}
+}
+
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+	logger.Info("successfully started")
+
+	// Watch for changes on the OBC CRD object
+	err = c.Watch(&source.Kind{Type: &bktv1alpha1.ObjectBucketClaim{TypeMeta: controllerTypeMeta}}, &handler.EnqueueRequestForObject{}, opcontroller.WatchControllerPredicate())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Reconcile reads that state of the cluster for a ObjectBucketClaim object and makes changes based on the state read
+// and the ObjectBucketClaim labels
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileOBCLabels) Reconcile(context context.Context, request reconcile.Request) (reconcile.Result, error) {
+	// workaround because the rook logging mechanism is not compatible with the controller-runtime logging interface
+	reconcileResponse, err := r.reconcile(request)
+	if err != nil {
+		logger.Errorf("failed to reconcile: %v", err)
+	}
+
+	return reconcileResponse, err
+}
+
+func (r *ReconcileOBCLabels) reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// Fetch the ObjectBucketClaim instance
+	cephOBC := &bktv1alpha1.ObjectBucketClaim{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, cephOBC)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			logger.Debugf("ObjectBucketClaim: %q resource not found. Ignoring since resource must be deleted.", request.NamespacedName)
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, errors.Wrapf(err, "failed to get ObjectBucketClaim: %q", request.NamespacedName)
+	}
+
+	// find the namespace for the ceph cluster (may be different then the namespace of the topic CR)
+	clusterNamespace := os.Getenv(k8sutil.PodNamespaceEnvVar)
```

Comment:
Using the operator namespace is not reliable. I'd just call out for the CephClusterList object and get the namespace from the object you find, typically one.

Reply:
can you point me to a place where we get the ceph cluster list without a namespace?
I see this kind of code that needs a namespace:
```
clusterList := &cephv1.CephClusterList{}
err := c.client.List(context.TODO(), clusterList, client.InNamespace(c.namespace))
```

Reply:
You can just pass `&client.ListOptions{}`, 


```go
	cephClusters := &cephv1.CephClusterList{}
	err := r.client.List(context.TODO(), cephClusters, &client.ListOptions{})
```

Reply:
but what do i do if i get more than one cluster?

Reply:
For some components, it's a known corner case but now that I think about it, looking at the op namespace is not such a bad idea. I guess we could do both. If we find something in the op ns we continue, if nothing we try to list all of them. If we find one we are good, if we find more than one we have a problem...